lib/schema.js

```diff
@@ -0,0 +1,92 @@
+const _ = require('lodash')
+const { joiStringRegexConvertible } = require('./joi-extension')
+const Joi = require('joi').extend([joiStringRegexConvertible])
+
+const category = Joi.object().keys({
+  title: Joi.string().required(),
+  label: Joi.string()
+})
+
+const searchReplace = Joi.object().keys({
+  search: Joi.stringRegexConvertible(),
+  replace: Joi.string()
+    .allow('')
+    .required()
+})
+
+const fields = {
+  branches: Joi.array()
+    .items(Joi.string())
+    .single()
+    .description(
+      'The branches to listen for configuration updates to `.github/release-drafter.yml` and for merge commits. ' +
+        'Useful if you want to test the app on a pull request branch. Default is the repository’s default branch.'
+    ),
+
+  'change-template': Joi.string().description(
+    'The template to use for each merged pull request.'
+  ),
+
+  'no-changes-template': Joi.string().description(
+    'The template to use for when there’s no changes. Default: `"* No changes"`.'
+  ),
+
+  'version-template': Joi.string().description(
+    'The template to use when calculating the next version number for the release. ' +
+      'Useful for projects that don’t use semantic versioning. ' +
+      'Default: `"$MAJOR.$MINOR.$PATCH"'
+  ),
+
+  'name-template': Joi.string().description(
+    'The template for the name of the draft release.'
+  ),
+
+  'tag-template': Joi.string().description(
+    'The template for the tag of the draft release.'
+  ),
+
+  'exclude-labels': Joi.array()
+    .items(Joi.string())
+    .description('Exclude pull requests using labels.'),
+
+  replacers: Joi.array()
+    .items(searchReplace)
+    .description('Search and replace content in the generated changelog body.'),
+
+  categories: Joi.array()
+    .items(category)
+    .description('Categorize pull requests using labels.'),
+
+  template: Joi.string()
+    .required()
```

Comment:
In the eyes of the one using the JSON schema outside the application, the template is not required. Since they could be using probot's `_extends` feature